### PR TITLE
Documentation and Packaging update (Pre mc/dc 0.0.9)

### PR DESCRIPTION
**Pull request by jpmorgan98**
# Changes:
- Documentation
- Paper
- Github Action CIs
- Packaging

## Documentation
- I added a contributors and users guide and made edits to reflect the changes 
- Added links on the side bar
- Added MC/DC logo and logos of our institutions to the home page

## Paper
I wrote a paper for JOSS submission and added the github workflow to compile it. The runner will be turned off after all updates are in. Please look over and see

## Packaging
Packaging is provided through [pip](https://pypi.org/project/mcdc/). This required a new `.toml` file and some creation of new accounts and API tokens to make it all work the CI runner which will deploy when a new version is detected. I also set up zenodo repo. Please note the new minimum requirements listed in the `.toml` file.

### New build options
Due to the significant size of `netgen` (~300MB) and its dependencies I added visualization as an optional build. This can installed by
```
pip install mcdc[viz]
```
To make the dependencies optional I put the `import`s in a `try` and `except` phraseology with a warning message when going to install

I also configured it to install the documentation dependencies in the same manner
```
pip install mcdc[doc]
```

##  Github CIs
When `pip install .` is run in a local directory all the necessary dependence will be snagged as part of that operation. So I disabled those in the github runner CIs. I also included 2 new runners to build the paper and package mcdc for pypi.

## README.md
I made significant changes to the `README.md` file to describe these changes

# Next steps
After this PR is accepted I will release version `0.9.0` which will provide a final update to pip for the package
I will fix some links to the documentation.
I will submit that paper to JOSS

### CODE REVIEW

**Comment by jpmorgan98** [comment]
Please note I am an awful speller while I do have spell check in VS code, someone should really give my copy a once over.

File: mcdc/input_.py
Comment (clemekay) [inline mcdc/input_.py:1]: Changes to mcdc/input_.py are in the docstrings of some functions to fix some sphinx issues.

**Review by clemekay — APPROVED** [review]

**Review by ilhamv — APPROVED** [review]

**Comment by ilhamv** [comment]
@jpmorgan98, let's merge it in after we are back in `black`. Great work!

**Comment by jpmorgan98** [comment]
Ok we should be good note that it looks like some stuff in the black standard has changed and you might need to update your version to get it to work `pip install black==24.1.0`

**Pull request merged**